### PR TITLE
scripts: twister: output error message

### PR DESCRIPTION
**Pull request by wangnuannuan**
Sometimes custom script fails, but we can't see the
error information in twister log, because the twister
uses _ to ignore the stderr.

Signed-off-by: Jingru Wang <jingru@synopsys.com>

### CODE REVIEW

**Review by PerMac — DISMISSED** [review]

File: scripts/pylib/twister/twisterlib.py
Comment (gopiotr) [inline scripts/pylib/twister/twisterlib.py:729]: Please consider f-string formatting, like this:
```
logger.error(f"Custom script failure: {stderr.decode(errors='ignore')}")
```
Advantages of that formatting can be found here:
https://realpython.com/python-f-strings/
https://realpython.com/python-string-formatting/

**Review by mbolivar-nordic — DISMISSED** [review]

**Review by gopiotr — DISMISSED** [review]

**Review by chen-png — DISMISSED** [review]

File: scripts/pylib/twister/twisterlib.py
Comment (temap) [reply scripts/pylib/twister/twisterlib.py]: @gopiotr logging lib is optimized to use the %s [formatting style](https://docs.python.org/3/howto/logging.html#optimization). Usually, pylint complains about f-string interpolation in logging.

Currently, the pre-script is silently falling without producing any info.
Is it the only reason the PR is not merged yet?

**Review by mbolivar-nordic — APPROVED** [review]

File: scripts/pylib/twister/twisterlib.py
Comment (mbolivar-nordic) [reply scripts/pylib/twister/twisterlib.py]: No, it's not approved because 1. the merge window was closed and 2. the reviewers currently have to re-approve after every push due to a policy change in the github project configurations. I've approved, but someone else needs to approve again also before we can merge

Comment (temap) [reply scripts/pylib/twister/twisterlib.py]: @mbolivar-nordic, thanks for the explanation.

**Review by nashif — APPROVED** [review]

File: scripts/pylib/twister/twisterlib.py
Comment (gopiotr) [reply scripts/pylib/twister/twisterlib.py]: @temap Thank you for letting me know about this optimization :) If somebody is more interested in this subject I recommend also this article: [What is the best string formatting technique for logging in Python?](https://dev.to/izabelakowal/what-is-the-best-string-formatting-technique-for-logging-in-python-d1d)

**Pull request merged**
